pyfakefs/fake_filesystem.py

```diff
@@ -2476,6 +2476,8 @@ def __init__(self, file_object, file_path, update=False, read=False, append=Fals
       contents = contents.recover(binary)
     if sys.version_info >= (3, 0):
       if binary:
+        io_class = io.BytesIO
+        # For Python 3, files opened as binary only read/write byte contents.
         if contents and isinstance(contents, str):
```

Comment:
Quite a mystery - I could have sworn that these 2 lines have never been deleted locally, so I couldn't understand  why the tests failed in GitHub...